### PR TITLE
2.0.16

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,12 @@ authors = ["GroovieGermanikus <groovie@mango.markets>"]
...
 [dependencies]
-yellowstone-grpc-client = { version = "1.13.0+solana.1.17.15", git = "https://github.com/rpcpool/yellowstone-grpc.git", tag = "v1.12.0+solana.1.17.15" }
-yellowstone-grpc-proto = { version = "1.12.0+solana.1.17.15", git = "https://github.com/rpcpool/yellowstone-grpc.git", tag = "v1.12.0+solana.1.17.15" }
+yellowstone-grpc-client = { version = "2.0.0", git = "https://github.com/rpcpool/yellowstone-grpc.git", tag = "v2.0.0+solana.2.0.16" }
+yellowstone-grpc-proto = { version = "2.0.0", git = "https://github.com/rpcpool/yellowstone-grpc.git", tag = "v2.0.0+solana.2.0.16" }
 
 
 # required for CommitmentConfig
-solana-sdk = "~1.17.15"
+solana-sdk = "2.0.16"
```
Comment (godmodegalactus) [inline Cargo.toml]: use `~2.0.16`

File: ci/rust-version.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@
 if [[ -n $RUST_STABLE_VERSION ]]; then
   stable_version="$RUST_STABLE_VERSION"
 else
-  stable_version=1.70.0
+  stable_version=1.81.0
```
Comment (godmodegalactus) [inline ci/rust-version.sh]: why 1.81.0 ? Should depend on the rust version of solana instead. I think it should be 1.78.0

File: Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "geyser-grpc-connector"
-version = "0.10.1+yellowstone.1.12"
+version = "2.0.0"
```
Comment (godmodegalactus) [inline Cargo.toml]: should have same naming convention

File: src/grpc_subscription_autoreconnect_streams.rs [hunk 4]
```diff
@@ -45,21 +46,18 @@ pub fn create_geyser_reconnecting_stream(
...
-                                    connect_timeout,
-                                    request_timeout,
-                                false)
-                                .await;
-                            let mut client = connect_result?;
+                            let mut builder =  GeyserGrpcClient::build_from_shared(addr).unwrap()
+                                .x_token(token).unwrap()
+                                .connect_timeout(connect_timeout.unwrap_or(Duration::from_secs(10)))
+                                .timeout(request_timeout.unwrap_or(Duration::from_secs(10)))
+                                .tls_config(config.unwrap_or(ClientTlsConfig::new())).unwrap();
```
Comment (godmodegalactus) [inline src/grpc_subscription_autoreconnect_streams.rs]: There is no tls config, check it again if new tls config means empty.

**Review by godmodegalactus — COMMENTED** [review]
I think it is missing some optimizations about the buffer sizes that groovie did.
@grooviegermanikus Could you check. Other than LGTM.

File: ci/rust-version.sh (hunk 2 above)
Comment (ghost) [reply ci/rust-version.sh]: The agave-xyz repository is on rust 1.81

File: Cargo.toml (hunk 3 above)
Comment (ghost) [reply Cargo.toml]: Can you eleborate on this naming convention - I don't understand what 0.10.1+yellowstone.1.12 means

File: src/grpc_subscription_autoreconnect_streams.rs (hunk 4 above)
Comment (ghost) [reply src/grpc_subscription_autoreconnect_streams.rs]: I ran the examples and they work without tls

File: Cargo.toml (hunk 1 above)
Comment (ghost) [reply Cargo.toml]: Why?

File: Cargo.toml (hunk 3 above)
Comment (grooviegermanikus) [reply Cargo.toml]: 0.10.1 is the version of the connector. "1.12" is the yellowstone-proto (protobuf) wire format version

File: Cargo.toml (hunk 1 above)
Comment (grooviegermanikus) [reply Cargo.toml]: you can look here https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html (+ assume that it's done on purpose ;) )

File: Cargo.toml [hunk 5]
```diff
@@ -32,6 +31,20 @@ bincode = "1.3.3"
 
 csv = "1.3.0"
 
+dashmap = "5.5.3"
+
+tonic = { version= "0.12.3", features=["gzip"] }
```
Comment (grooviegermanikus) [inline Cargo.toml:35]: tonic dependency had some historic version issues. do we need that here?

File: ci/rust-version.sh (hunk 2 above)
Comment (grooviegermanikus) [reply ci/rust-version.sh]: disagree, look here -> https://github.com/anza-xyz/agave/blob/v2.0.16/rust-toolchain.toml

**Review by grooviegermanikus — CHANGES_REQUESTED** [review]
please look at the Cargo.toml version issue

File: Cargo.toml (hunk 1 above)
Comment (godmodegalactus) [reply Cargo.toml]: This library is used my multiple projects which would have different minor version. We want to not update this version all the time for different projects. In anycase replacing this version with other minor version upgrade should not break anything in this library.

File: ci/rust-version.sh (hunk 2 above)
Comment (ghost) [reply ci/rust-version.sh]: Yeah, you are right - I was looking the main branch...

File: Cargo.toml (hunk 1 above)
Comment (ghost) [reply Cargo.toml]: > you can look here https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html (+ assume that it's done on purpose ;) )

I know what the tilde requirements does. I would rather do proper dependency management and make sure that my projects use the same dependencies. But, I see where you are coming from.

File: Cargo.toml (hunk 3 above)
Comment (ghost) [reply Cargo.toml]: Okay that sounds like we want 0.11.0+yellowstone.2.0.0 for this

File: Cargo.toml (hunk 5 above)
Comment (ghost) [reply Cargo.toml:35]: @grooviegermanikus  - I think those dependencies cam from your performancetest branch - I just updated tonic to most recent - we need this to enable gzip in yellowstone_grpc_util

```rust
        GeyserClient::with_interceptor(channel, interceptor)
            .accept_compressed(CompressionEncoding::Gzip)
```

**Comment by mschneider** [comment]
I updated traffic and can confirm this does build, we should merge this branch, once we verified the whole integration traffic + connector + grpc + rpc 2.0.16 works. Sent @mango-dee creds for setting up a fresh rpc in frankfurt.

File: Cargo.toml (hunk 5 above)
Comment (mschneider) [reply Cargo.toml:35]: lets move forward, i found a workaround for the zeroize version issue.

File: Cargo.toml (hunk 3 above)
Comment (mschneider) [reply Cargo.toml]: would prefer 0.11.0+solana-2.0.16 or 0.11.0+yellowstone-2.0.0+solana-2.0.16, the solana dependency is the main thing people will want to look out for when using this package, but if there's a good reason for yellowstone tags, then let's do both. yellowstone-2.0.0 unfortunately supports many solana versions

File: Cargo.toml (hunk 5 above)
Comment (grooviegermanikus) [reply Cargo.toml:35]: you can drop Gzip support if it helps

**Review by grooviegermanikus — APPROVED** [review]

File: Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "geyser-grpc-connector"
-version = "0.10.1+yellowstone.1.12"
+version = "0.11.0+yellowstone.2.0.0"
```
Comment (grooviegermanikus) [inline Cargo.toml:3]: Yellowstone proto is 1.15 not 2.0.0 afaik

**Pull request closed without merging**